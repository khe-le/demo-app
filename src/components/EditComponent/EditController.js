import React, { useState, useEffect } from 'react'
import EditList from './EditList'
import EditForm from './EditForm'
import IslandStore from '../actions/store'

function EditController(){
    const [villagers, setVillagers] = useState(IslandStore.initialState)
    const [editorDetail, setEditorDetail] = useState(false)
    const [editorState, setEditorState] = useState(false)

    useEffect(()=> {
            const subs = IslandStore.subscribe(setVillagers);
            IslandStore.init();
        
            return () => subs.unsubscribe();
        },[]);
    
    function removeVillager(villager){
        IslandStore.removeVillager(villager)
        IslandStore.init();
        setEditorState(false)
    }

    function openEditor(villager){
        setEditorState(true)
        setEditorDetail(villager)
    }

    function closeEditor(){
        setEditorState(false)
    }

    function updateVillager(villager){
        IslandStore.updateVillager(villager)
        IslandStore.init();
    }

    return (
        <React.Fragment>
            <EditList
                removeVillager={removeVillager}
                villagers={villagers.data}
                openEditor={openEditor}
            />
            {editorState ? 
                <EditForm 
                            updateVillager={updateVillager}
                            editVillager={editorDetail}
                            closeEditor={closeEditor}
                            /> : null}
        </React.Fragment>
    )
}

export default EditController