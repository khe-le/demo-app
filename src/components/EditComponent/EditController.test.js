import React from 'react'
import EditController from './EditController'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('<EditController />', () => {
    let wrapper = Enzyme.shallow(<EditController/>)

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
})