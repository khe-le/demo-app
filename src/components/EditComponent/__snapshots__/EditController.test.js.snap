// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditController /> should match the snapshot 1`] = `"<div class=\\"villager-list\\"></div>"`;
